### ISSUE TITLE
Gotify claims to be connected when server is shut down

### ISSUE THREAD

**Issue by clefru** [gotify/android#319]
Gotify in the status bar claims that it is "connected" after several minutes despite me shutting down the server. The default config contains `stream: pingperiodseconds: 45`, so I presume that there is some keep-alive mechanism that detects failure. It would nice, if gotify would notify me in case the connection goes down.

**Comment by jmattheis** [comment]
Gotify should already do this, please post your gotify/server config, reverse proxy config and gotify/server log.

Does the android app error if the server is down and you manually refresh the app here?
![image](https://github.com/gotify/android/assets/14895212/ddc9327b-b23e-4a75-81d6-0314f43801bd)

**Comment by clefru** [comment]
Thanks for the extremely quick response. I retried the "kill the server" experiment today to answer your questions, and to my surprise gotify on Android correctly shows "Not connected" almost immediately after I kill the server.

[Here is the log of the app from my yesterday's experiment](https://gist.github.com/clefru/39e53c54f89635816833fb2c5e617412). I think that I spotted something odd in there:
* I created my DNS entry for the server shortly before logging in on my Android phone. The hostname is gotify.endorphin.org.
* Around 17:23, it was already resolving as can be seen in this request line:
```2023-10-13 17:23:18.203 ERROR: Error while api call: com.github.gotify.api.ApiException: Request failed.
[..]
Caused by: java.net.ConnectException: Failed to connect to gotify.endorphin.org/146.148.99.138:443
```
* But some time later, the hostname went back to being non-resolvable:
* 
```
2023-10-13 18:09:13.587 ERROR: WebSocket(20): failure  Message: : java.net.UnknownHostException: Unable to resolve host "gotify.endorphin.org": No address associated with hostname
```

Maybe my DNS provider didn't do a good job updating all of their nameservers simultaneously? I can only speculate as this isn't easy to reproduce.

I am sorry for the noise. I think that I will simply close this bug, because DNS issues are probably too much of a corner case for anyone to debug in detail.
